lib/uwb/include/uwb/UwbDevice.hxx

```diff
@@ -51,6 +51,14 @@ public:
     void
     Reset();
 
+    /**
+     * @brief Initializes a new UWB device
+     * 
+     * @return bool
+     */
+    virtual bool
+    Initialize();
```

Comment:
This should follow the existing pimpl pattern where the top-level `UwbDevice` object defines the interface behavior with a non-virtual function (here, `bool Initialize()`) and then declares a private virtual impl function that that derived classes implement (here, `bool InitializeImpl`). If the derived class is required to provide such an impl, then it should be pure virtual (`eg. bool InitializeImpl() = 0`), but in this case it makes sense not to make it pure since some derived impls wouldn't need to initialize anything.

This pattern helps keep the interface stable and allows us to instrument operations from the highest level in the class hierarchy.